GOVERNANCE.md

```diff
@@ -0,0 +1,176 @@
+* Robert Milan ([Grafana Labs](https://grafana.com)) (robert@grafana.com)
+
+### Technical decisions
+
+Technical decisions should be discussed and voted upon on the Metrictank [development mailing list][devlist]. Informal conversations may also be held in *#metrictank-dev* in [slack][slack] or in the appropriate Github issue.
+
+Decisions are usually made by [lazy consensus](#consensus). If no consensus can be reached, the matter may be resolved by [majority](#majority-vote) vote.
+
+### Governance changes
+
+Material changes to this document are discussed publicly in the *#metrictank-dev* channel in [slack][slack] or on the Metrictank [development mailing list][devlist]. Any change requires a [supermajority](#supermajority-vote) in favor.
+
+### Editorial changes
+
+Editorial changes are changes which fix spelling or grammar, update work affiliation, or similar; they update style or reflect an outside and obvious reality. They do not change the intention or meaning of anything in this document. They must be made via PR and accepted via [lazy consensus](#consensus).
+
+### Other matters
+
+Any matter that needs a decision, including but not limited to financial matters, may be called to a vote by any member if they deem it necessary. For financial, private, or personnel matters, discussion and voting takes place on the private [team mailing list][team], otherwise on the Metrictank [development mailing list][devlist].
+
+## Voting
+
+The Metrictank project usually runs by [lazy consensus](#consensus), however sometimes a formal decision must be made.
+
+Depending on the subject matter, as laid out above, different methods of voting are used.
+
+For all votes, including [lazy consensus](#consensus), voting must be open for at least 21 days. The end date should be clearly stated in the call to vote. A vote may be called and closed early if enough votes have come in one way so that further votes cannot change the final decision.
+
+In all cases, all and only team members are eligible to vote, with the sole exception of the forced removal of a team member, in which said member is not eligible to vote.
+
+Discussions and votes on personnel matters (including but not limited to team membership) are held on the private [team mailing list][team]. All other discussions are held in the *#metrictank-dev* channel in [slack][slack], in their respective Github issues, or on the Metrictank [development mailing list][devlist]. All other votes are held on the Metrictank [development mailing list][devlist].
+
+For public discussions, anyone interested is encouraged to participate. Formal power to object or vote is limited to team members.
+
+### Consensus
+
+The default decision making mechanism for the Metrictank project is lazy consensus, which is itself a form of voting. This means that any decision on technical issues is considered supported by the team as long as nobody objects.
```

Comment:
When is any PR a "technical issue on which a decision should be made" ?
If all/most of them are, I just realized that this may significantly slow down our process.
This would require either explicit signoff from all the team members or expiration of the 21day timeout, until we can merge PR's.

That doesn't seem practical.

Reply:
If we only have 5 team members then this would require explicit signoff from 3 for majority or 4 for supermajority, at which point the last person's vote doesn't matter.

Maybe we should change it back to 14 days then, or try to better define categories of changes. It's a bit dangerous to get into specifics though because it is a very grey area.

That being said, I don't think voting will affect many PRs. Voting and consensus are around the idea to change something, not necessarily the actual change. Once a decision has been made / voted upon and a PR is opened do we want to vote on that PR then? If not, then as long as the issue or thread in the dev mailing list already passed with either lazy consensus or a formal vote, there is no need to use this process.

Reply:
can you find out how this is dealth with in other projects?

Reply:
I like the new wording around PR's. thanks